packages/application-shell/src/utils/http-client.ts

```diff
@@ -197,6 +197,7 @@ function createHttpClientOptions(config: TConfig = {}): TOptions {
         ? `Bearer ${sessionToken}`
         : undefined,
       [SUPPORTED_HEADERS.X_APPLICATION_ID]: window.app.applicationId,
+      [SUPPORTED_HEADERS.X_CUSTOM_VIEW_ID]: window.app.customViewId,
```

Comment:
I'm only wondering if we should send the header if `customViewId` is undefined?
```suggestion
      ...(window.app.customViewId
        ? { [SUPPORTED_HEADERS.X_CUSTOM_VIEW_ID]: window.app.customViewId }
        : {}),
```

Reply:
Thanks Kacper 👍 

I had the same thought at first, but notice this is inside a call to the `omitEmpty` helper function which already would remove it in such case.